
# HG changeset patch
# User Andreas Regel <andreas.regel@gmx.de>
# Date 1270476788 -7200
# Node ID d097ecaa55ea1e7ca6296d7dcb73df6c3d3a7c90
# Parent e268d8395600fe17a5d498b80014be1831e59592
stb0899: set FE_HAS_SIGNAL flag in stb0899_read_status

Signed-off-by: Andreas Regel <andreas.regel@gmx.de>

--- a/linux/drivers/media/dvb/frontends/stb0899_drv.c	Mon Apr 05 15:45:21 2010 +0200
+++ b/linux/drivers/media/dvb/frontends/stb0899_drv.c	Mon Apr 05 16:13:08 2010 +0200
@@ -1119,7 +1119,7 @@
 			reg  = stb0899_read_reg(state, STB0899_VSTATUS);
 			if (STB0899_GETFIELD(VSTATUS_LOCKEDVIT, reg)) {
 				dprintk(state->verbose, FE_DEBUG, 1, "--------> FE_HAS_CARRIER | FE_HAS_LOCK");
-				*status |= FE_HAS_CARRIER | FE_HAS_LOCK;
+				*status |= FE_HAS_SIGNAL | FE_HAS_CARRIER | FE_HAS_LOCK;
 
 				reg = stb0899_read_reg(state, STB0899_PLPARM);
 				if (STB0899_GETFIELD(VITCURPUN, reg)) {
@@ -1136,7 +1136,7 @@
 		if (internal->lock) {
 			reg = STB0899_READ_S2REG(STB0899_S2DEMOD, DMD_STAT2);
 			if (STB0899_GETFIELD(UWP_LOCK, reg) && STB0899_GETFIELD(CSM_LOCK, reg)) {
-				*status |= FE_HAS_CARRIER;
+				*status |= FE_HAS_SIGNAL | FE_HAS_CARRIER;
 				dprintk(state->verbose, FE_DEBUG, 1,
 					"UWP & CSM Lock ! ---> DVB-S2 FE_HAS_CARRIER");
 

